src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const listOfStates = [];
+  const newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(newState, action.extraData);
+        listOfStates.push({ ...newState });
```

Comment:
Instead of pushing a new object copy into the 'listOfStates' inside each case, you can perform this action once after the switch statement to follow the DRY (Don't Repeat Yourself) principle.